src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java

```diff
@@ -903,5 +903,50 @@
      *        Enclave attestation protocol.
      */
     void setEnclaveAttestationProtocol(String protocol);
+    
+    /**
+     * Returns client certificate path for client certificate authentication.
+     * 
+     * @return Client certificate path.
+     */
+    String getClientCertificate();
+
+    /**
+     * Sets client certificate path for client certificate authentication.
+     * 
+     * @param certPath
+     *        Client certificate path.
+     */
+    void setClientCertificate(String certPath);
+    
+    /**
+     * Returns Private key file path for client certificate authentication.
+     * 
+     * @return Private key file path.
+     */
+    String getClientKey();
+
+    /**
+     * Sets Private key file path for client certificate authentication.
+     * 
+     * @param keyPath
+     *        Private key file path.
+     */
+    void setClientKey(String keyPath);
+    
+    /**
+     * Returns the password to be used for Private key provided by the user for client certificate authentication.
+     * 
+     * @return Private key password.
+     */
+    String getClientKeyPassword();
```

Comment:
should we have this? we don't usually have getters for passwords

Reply:
Good point, I'll remove this one.